### PR TITLE
CI and test optimizations

### PR DESCRIPTION
**Pull request by dewyatt**
This does a few things with the primary goal of speeding up CI. It also speeds up local testing quite a bit on my system.

From a developer usage standpoint, the main things to be aware of are these new ways of running tests:
```sh
ctest -R rnp_tests                     # run all of rnp_tests
ctest -R rnp_tests-test_dsa_roundtrip  # run this one particular test in rnp_tests
ctest -R cli_tests                     # run all cli_tests.py test suites
ctest -R cli_tests-EncryptSignRSA      # run the EncryptSignRSA test suite in cli_tests.py
ctest -R ruby-rnp                      # run ruby-rnp tests
```

(and if you want things to run fast, make sure you're using `-j`)

### CODE REVIEW

**Comment by dewyatt** [comment]
Oh I should also mention, if you add a new suite to `cli_tests.py`, you'll probably have to re-run CMake.

**Comment by dewyatt** [comment]
Looks like I'm going to be running into https://github.com/riboseinc/rnp/pull/867 again for macOS. Hmm.

**Comment by ronaldtse** [comment]
@dewyatt seems that there is one failure on Cirrus with ruby-rnp:

```
The following tests FAILED:
	131 - ruby-rnp (Failed)
Errors while running CTest
```

On the other hand, for the new practices:

> ```sh
> ctest -R rnp_tests                     # run all of rnp_tests
> ctest -R rnp_tests-test_dsa_roundtrip  # run this one particular test in rnp_tests
> ctest -R cli_tests                     # run all cli_tests.py test suites
> ctest -R cli_tests-EncryptSignRSA      # run the EncryptSignRSA test suite in cli_tests.py
> ctest -R ruby-rnp                      # run ruby-rnp tests
> ```

and 

> (and if you want things to run fast, make sure you're using `-j`)

and

> if you add a new suite to cli_tests.py, you'll probably have to re-run CMake.

Can we update the repo's development guide, and possibly establish one on https://github.com/riboseinc/rnpgp.com to contain description of how people can work with RNP development?

Thanks!

File: cmake/CTestCostData.txt [hunk 1]
```diff
@@ -0,0 +1,132 @@
+rnp_tests-hash_test_success 3 0.0164573
```
Comment (ni4) [inline cmake/CTestCostData.txt]: Do I correctly understand that we should update this file with each new test added?

Comment (dewyatt) [reply cmake/CTestCostData.txt]: It's not necessarily beneficial to update it on every added test. It's more useful to update it occasionally or particularly if adding a long-running test. Tests that aren't listed here will default to a COST of 0 (likely to start running last).

I need to add this to the dev guide.

**Comment by dewyatt** [comment]
> @dewyatt seems that there is one failure on Cirrus with ruby-rnp:
> 
Should be fixed now, we'll see.

 > Can we update the repo's development guide, and possibly establish one on https://github.com/riboseinc/rnpgp.com to contain description of how people can work with RNP development?
> 

I do need to update the dev guide. It would probably be good to add a "Getting Involved"/"Contributing" type of page on the website too.

**Comment by ronaldtse** [comment]
@dewyatt there's a conflict here, could you help fix and directly merge this? Thanks!

**Comment by dewyatt** [comment]
Great, something in the rebase broke everything.

**Comment by dewyatt** [comment]
Filed an issue: https://github.com/riboseinc/rnp/issues/895

It's affecting this PR but is also present in master (doesn't manifest in master unless you disable all other tests in rnp_tests).

**Comment by ni4** [comment]
@dewyatt I guess this should be rebased to let all tests pass?

**Comment by dewyatt** [comment]
@ni4 Yep, will rebase.

**Comment by dewyatt** [comment]
Merging as requested above!

**Comment by ronaldtse** [comment]
Thanks @dewyatt !

**Pull request merged**